src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
+1. `MarkLoops` - performs an analysis of `Expressions'` connectivity and their `PortDescriptors`. Based on this information, the pass divides the `Expression` into groups, so that each of the groups can be executed inside one loop. Every group is described by a `loop_id`, and additional information is saved in `LinearIR::LoopManager::LoopInfo`.
+2. `FuseLoops` - analyzes the assigned `loop_ids` and `LoopInfo`, fuses some loops if possible. This pass can move some `Expressions` up or down the graph, so the `Expressions` with the same `loop_ids` are grouped together.
+3. `InsertBuffers` - analyzes `LoopInfo` and `Expression` semantics, inserts `snippets::op::Buffer`. `Buffer` is an operation that represents a memory buffer needed to save some intermediate results.
+4. `InsertLoadStore` - inserts explicit memory access operations like `Load` and `Store` (both from `snippets::op`). These are needed, so appropriate instructions will be emitted during the code generation stage to move data between memory and vector registers.
+5. `InitLoops` - inserts explicit `LoopBeegin` and `LoopEnd` (`snippets::op`) operations based on the acquired `LoopInfo`. Again, the explicit operations are needed to emit appropriate instructions later.
+6. `LoadMoveBroadcastToBroadcastLoad` fuse `Load->MoveBroadcast` sequences into a single `BroadcastLoad` expressions.
 
-* nGraph operation (for example, `class FMA : public ov::op::Op`)
-* Emitter for the operation (for example, `class FmaEmitter : public Emitter` )
-* register the pair in `jitters` map
+The buffer pipeline is focused on managing `op::Buffer` operations:
+1. `IdentifyBuffers` - analyzes buffers using graph coloring algorithm to enable memory reuse.
+2. `AllocateBuffers` - calculates minimal memory size required to handle all the buffers. The memory will be allocated during the execution stage by the plugin `Snippet` node.
+    
+The final pipeline applies finishing touches before the `IR` will be passed to the `Generator` module:
+1. `PropagateLayout` - propagates data layouts to `Parameters` and `Results` (actually to corresponding `Expressions`), so `Kernel` will be able to calculate appropriate data offsets for every iteration of an external parallel loop.
+2. `CleanupLoopOffsets` - eliminates redundant pointer increments from the loops.
 
-### Calling convention
+More details on control flow optimization passes could be found in the `control_flow_transformations(...)` method inside [subgraph.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/op/subgraph.cpp). When all the passes are applied, the `LinearIR` is handled further to the `Generator` to emit executable code.
```

Comment:
BTW you can use relative links in markdown to avoid the absolute paths in network.
Also you can use reference links (to reuse them, for example)

For [more](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links) information.

Reply:
Thanks for the suggestion! will do